### PR TITLE
[TF API] Enable single Dataset type

### PR DESCRIPTION
**Pull request by rxwei**
`#tfop` now supports aggregate packing and unpacking. We merge `SingleValueDataset` and `DoubleValueDataset` to a single type `Dataset<Element>` with unconstrained `Element`.

After we switch to the new type, it's significantly harder and more ad-hoc to form generic conversion initializers from Numpy. As such, Numpy conversion initializers should be added to our tutorial code as hidden source for now.

**Known issues**:
- [SR-8684](https://bugs.swift.org/browse/SR-8684): Higher-order functions are problematic and would trigger an assertion in SILGen on function conversion.

### CODE REVIEW

**Comment by rxwei** [comment]
@swift-ci please test tensorflow Linux CPU

**Comment by rxwei** [comment]
@swift-ci please test tensorflow Linux GPU

**Comment by rxwei** [comment]
@swift-ci please test tensorflow Linux GPU

**Review by dan-zheng — APPROVED** [review]
LGTM, nice generalization over arities (in the absence of variadic generics)!

File: stdlib/public/TensorFlow/Dataset.swift [hunk 1]
```diff
@@ -62,329 +54,151 @@ public struct SingleValueDataset<ScalarOfElement : AccelerableByTensorFlow> {
...
+public extension Dataset {
   @inlinable @inline(__always)
   init(randomSeed: Int64) {
     let (seed1, seed2) = _tensorSeeds(Tensor(randomSeed))
     self.init(
       _handle: #tfop("RandomDataset", seed1, seed2,
-                     output_types: [ScalarOfElement.self],
-                     output_shapes: [nil] as [TensorShape?])
+                     output_types$array: Element.self,
+                     output_shapes$unknownShapeList: Element.self)
```
[the remark this answers is not in the document]
Comment (rxwei) [reply stdlib/public/TensorFlow/Dataset.swift]: `nil` maps to `?` in TensorFlow. Shapes are then dynamically inferred.

**Pull request merged**